src/k8s-engine/wsl.ts

```diff
@@ -1207,8 +1230,29 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
             const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '')
               .replace('/var/log', logPath);
 
+            await this.writeFile('/etc/init.d/host-resolver', SERVICE_SCRIPT_HOST_RESOLVER, { permissions: 0o755 });
             await this.writeFile('/etc/init.d/dnsmasq-generate', SERVICE_SCRIPT_DNSMASQ_GENERATE, { permissions: 0o755 });
+            // we add both host-resolver & dnsmasq-generate services to run-level default
+            // we del as per config. This is to suppress the rc-update del error if a service
+            // is not previously added
+            await this.execCommand('/sbin/rc-update', 'add', 'host-resolver', 'default');
             await this.execCommand('/sbin/rc-update', 'add', 'dnsmasq-generate', 'default');
```

Comment:
Need to `rc-update add dnsmasq default` as well (that's a distinct service from `dnsmasq-generate`).